### PR TITLE
[IMP] pylint_odoo: Add check duplicate ids in ir.model.access.csv files

### PR DESCRIPTION
**Pull request by lescobarvx**
[IMP] pylint_odoo: Add check duplicate ids in ir.model.access.csv files

### CODE REVIEW

File: pylint_odoo/test_repo/broken_module/ir.model.access.csv [hunk 1]
```diff
@@ -0,0 +1,7 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
```
Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/ir.model.access.csv]: add a empty .csv in other module, please

Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/ir.model.access.csv]: add a duplicate id of different csv in same module

File: pylint_odoo/checkers/modules_odoo.py [hunk 2]
```diff
@@ -131,6 +136,21 @@ def _check_duplicate_xml_record_id(self):
...
+        :return: False if exists errors and
+                 add list of errors in self.msg_args
+        '''
+        all_csv_ids = []
+        for csv_file in self.filter_files_ext('csv', relpath=False):
+            os.path.basename(csv_file) == 'ir.model.access.csv' and \
+                all_csv_ids.extend(self.get_field_csv(csv_file))
+        duplicated_id_csv = self.get_duplicated_items(all_csv_ids)
+        if duplicated_id_csv:
+            self.msg_args = duplicated_id_csv
```
Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: Add the subpath of the file as args to get info:
`security/ir.model.access.csv`
`ir.model.access.csv`
`other_path/ir.model.access.csv`

File: pylint_odoo/misc.py [hunk 3]
```diff
@@ -213,3 +214,18 @@ def get_xml_record_ids(self, xml_file, module=None):
...
+    def get_field_csv(self, csv_file, field='id'):
+        '''Get xml ids from csv file
+        :param csv_file: Path of file csv
+        :param field: Field to search
+        :return: List of string with field rows
+        '''
+        csv_ids = []
+        with open(csv_file, 'rb') as csvfile:
+            lines = list(csv.reader(csvfile))
+            row = lines[0].index(field)
```
Comment (moylop260) [inline pylint_odoo/misc.py]: Will show a error in a empty file?

What about use `csv.dictreader`?

File: pylint_odoo/checkers/modules_odoo.py [hunk 4]
```diff
@@ -131,6 +136,21 @@ def _check_duplicate_xml_record_id(self):
...
 
+    def _check_duplicate_id_csv(self):
+        '''Check duplicate xml id in ir.model.access.csv files of a odoo module.
+        :return: False if exists errors and
+                 add list of errors in self.msg_args
+        '''
+        all_csv_ids = []
+        for csv_file in self.filter_files_ext('csv', relpath=False):
+            os.path.basename(csv_file) == 'ir.model.access.csv' and \
+                all_csv_ids.extend(self.get_field_csv(csv_file))
```
Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: pylint show `expression-not-assigned` here

File: pylint_odoo/test/main.py [hunk 5]
```diff
@@ -8,7 +8,7 @@
 from pylint_odoo import misc
 
 
-EXPECTED_ERRORS = 56
+EXPECTED_ERRORS = 58
```
Comment (moylop260) [inline pylint_odoo/test/main.py]: Use `EXPECTED_ERRORS += 2  # Check...`

File: pylint_odoo/test_repo/broken_module/ir.model.access.csv (hunk 1 above)
Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/ir.model.access.csv]: Add other csv file without `id` field, please

Comment (moylop260) [inline pylint_odoo/test_repo/broken_module/ir.model.access.csv]: Add other csv with ideal case

File: pylint_odoo/misc.py [hunk 6]
```diff
@@ -213,3 +214,17 @@ def get_xml_record_ids(self, xml_file, module=None):
...
+        '''Get xml ids from csv file
+        :param csv_file: Path of file csv
+        :param field: Field to search
+        :return: List of string with field rows
+        '''
+        csv_ids = []
+        with open(csv_file, 'rb') as csvfile:
+            lines = csv.DictReader(csvfile)
+            for line in lines:
+                if field in line.keys():
```
Comment (moylop260) [inline pylint_odoo/misc.py]: Use
`if line.has_key(field):`

Or use a list comprehensions:
`return [line[field] for line in lines if line.has_key(field)]`

**Comment by moylop260** [comment]
:+

**Comment by moylop260** [comment]
:+1:

**Comment by moylop260** [comment]
Could you create a cherry-pick to create a new PR for oca project?

**Comment by lescobarvx** [comment]
@moylop260
with pleasure

**Pull request merged**